src/manifold/stream/default.clj

```diff
@@ -38,6 +45,17 @@
     nil
     x))
 
+(defn- cleanup-expired-deferreds
+  "Removes all realized deferreds (presumably from timing out)."
+  [^LinkedList l]
+  (when-not (.isEmpty l)
```

Comment:
this should probably be synchronized on `l`, just to be safe

Reply:
Ahh, thanks for catching that, I'll update it.